src/averageAges.js

```diff
@@ -31,13 +46,25 @@ function calculateMenAverageAge(people, century) {
  * Hint: To check if a woman has children you should find someone who mention
  * her as mother.
  *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
+//  * @param {object[]} people
+//  * @param {boolean} withChildren - optional
+//  *
+//  * @return {number}
+//  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWoman = withChildren
+    ? people
+      .filter(person => (
+        people.find(child => child.mother === person.name && person.sex === 'f')
+      ))
+    : people
+      .filter(person => person.sex === 'f');
+
+  const averageAge = filteredWoman
+    .reduce((sum, { born, died }) => sum + (died - born), 0)
+      / filteredWoman.length;
```

Comment:
You do the same calculations with reduce in every function. Maybe we should move this calculation to a separate helper?